universe/Ship.cpp

```diff
@@ -456,6 +457,10 @@ namespace {
         int available_fighters = 0;
 
         retval.reserve(parts.size() + 1);
+        //const ScriptingContext context(ship); // XXX expects a shared_ptr ???
+        std::shared_ptr<Ship> ship_clone(ship->Clone()); // XXX ugly way to create a shared_ptr necessary for the context
+        const ScriptingContext context(ship_clone);
```

Comment:
@geoffthemedio is there a better way to get a shared_ptr for a ship so I can use it as Source in context evaluation?

Reply:
should be able to use `shared_from_this()`